src/core/components/InstallButton/index.js

```diff
@@ -72,6 +84,32 @@ export class InstallButtonBase extends React.Component {
           {i18n.gettext('Install Theme')}
         </Button>
       );
+    } else if (addon.type === ADDON_TYPE_OPENSEARCH) {
+      const onClick = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        if (
+          !buttonIsDisabled && _window.external &&
+          'AddSearchProvider' in _window.external
+        ) {
+          _log.info('Adding OpenSearch Provider', { addon });
+          _window.external.AddSearchProvider(addon.installURL);
```

Comment:
Sadly there's no callback or error event. But if the function doesn't exist it should never be called and it's an API that to my mind only Firefox implements. So it shouldn't ever fail... 🤷‍♀️